File: 首期/第一阶段/第二讲/ll.go
package main

import "fmt"

// Node 是一个元素
type Node struct {
	v    int
	next *Node
}

func (n *Node) add(v int) {
	if n.next == nil {
		n.next = &Node{v: v}
		return
	}
	cur := n.next
	for cur.next != nil {
		cur = cur.next
	}
	cur.next = &Node{v: v}
}

func (n *Node) print() {
	for cur := n; cur != nil; cur = cur.next {
		fmt.Println(cur.v)
	}
}

func main() {
	n := Node{v: 10}
	n.add(2)
	n.print()

	a := 10
	a = "xxx"
}

File: 首期/第一阶段/第二讲/ll.js
/*

类型强弱  -- 值的类型
类型动静  -- 变量名

Python  动态强类型
Go      静态强类型
JS      动态弱类型

*/
function LinkedList() {
    return {
        head: null,
        next: null,
        add: function(v) {
            if (!this.head) {
                this.head = new Node(v)
                return
            }
            let cur = this.head
            while (cur.next) {
                cur = cur.next
            }
            cur.next = new Node(v)
        },
        print: function() {
            // cur := n.next
            // for cur.next != nil {
            //     cur = cur.next
            // }
            // cur.next = &Node{v: v}
            let cur = this.head
            while ( cur ) {
                console.log(cur.v)
                cur = cur.next
            }
        }
    }
}

function Node(v) {
    return {
        v: v,
        next: null,
    }
}

// function Node(v) {
//     this.v = v
//     this.next = null
// }

let ll = LinkedList()
ll.add(10)
ll.add(-2)
ll.print()
